google/cloud/firestore_v1/base_client.py

```diff
@@ -117,11 +113,10 @@ def __init__(
         self._database = database
         self._emulator_host = os.getenv(_FIRESTORE_EMULATOR_HOST)
 
-    @property
-    def _firestore_api(self):
+    def _firestore_api_helper(self, transport, client_class, client_module):
         """Lazy-loading getter GAPIC Firestore API.
         Returns:
-            :class:`~google.cloud.gapic.firestore.v1`.firestore_client.FirestoreClient:
+            :class:`~google.cloud.gapic.firestore.v1`.async_firestore_client.FirestoreAsyncClient:
```

Comment:
Isn't the type returned here dependant upon the subclass, either Async or not?

Is there a way we can list the or in the type annotation?

Reply:
That is a good point, and I'm not sure what the best way to resolve this is. Give me a bit to get back to you on that.

Reply:
From what it looks like [here](https://pythonhosted.org/an_example_pypi_project/sphinx.html#function-definitions), type is not required.

Reply:
I decided to remove type entirely, as `base_client` has no requirement on return type for the function. Its conceivable that at some point a client other than async and sync may exist, in which case the return type we specify would be outdated.

Reply:
Fair, *grumbles about lack of scope constraints* :cry: